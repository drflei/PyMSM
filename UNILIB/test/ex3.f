      PROGRAM example3
C
      INCLUDE 'structure.h'
C
      INTEGER*4    kunit, kinit, ifail, kint, kext
      CHARACTER*32 lbint, lbext
      REAL*8       year, param(10), amjd
      INTEGER*4    knfl, ktyplus
      REAL*8       fbm0, flm0, falt, phi, star
C
C     initialize variables
C
      DATA kunit, kinit/ 6, 1/
      DATA kint, kext/ 0, 0/
      DATA year, param, amjd/ 1995.d0, 10*0.0d0, 0.0d0/
C
C     initialize the library
C
      CALL UT990 (kunit, kinit, ifail)
      IF( ifail .LT. 0 )STOP
C
C     set the geomagnetic field model
C     (kint=0, DGRF/IGRF)
C
      CALL UM510 (kint, year, lbint, kunit, ifail)
      IF( ifail .LT. 0 )STOP
C
C     set no external magnetic field model
C
C
      CALL UM520 (kext, amjd, param,
     :           lbext, kunit, ifail)
      IF( ifail .LT. 0 )STOP
C
C     trace the drift shell
C
      fbm0          =    0.19d0
      flm0          =    2.0d0
      falt          =    0.0d0
      knfl          =   60
      ktyplus       =    3
C
      CALL UD310 (fbm0, flm0, falt, knfl, 
     :           ktyplus, ifail)
      IF( ifail .LT. 0 )STOP
C
C     evaluate the third invariant
C
      CALL UD330 (phi, star, ifail) 
      IF( ifail .LT. 0 )STOP
C
C     write the result
C
      WRITE(kunit,1000) phi,star
C
 1000 FORMAT(
     :  /25('-'),/'RESULT:',
     :  /'Third invariant =', es24.14, ' ; Lstar =', es24.14,
     :  /25('-') )
C
      END

