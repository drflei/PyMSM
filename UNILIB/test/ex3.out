
                                    SPENVIS
                                 UNILIB Library
                                 Version 2.22_YCH_initPrep
                           Generated by SPENVIS team


 --- Main control parameters ---

      UC160 (general constants):
                      Geoid major axis = 6378.16     km
                      Geoid minor axis = 6356.77     km
                Mean Earth radius (Re) = 6371.20     km
                  Ecliptic inclination =   23.441512 deg
                 McIlwain Earth dipole =    0.311653 G/Re^3

      UC190 (field line tracing):
                     Maximum step size =  477.8      km at one Re
              Maximum "last" step size =   15.0      km
                     Minimum step size =    2.0      km
                    Altitude precision =  200.       m
              Magnetic field precision =    0.0006   %

      UC192 (magnetic field evaluation):
             Max. geomagnetic latitude =   85.0      deg
                        Minimum radius =    0.157    Re
            (drift shell tracing):
                        Precision on L =    0.10000  %
            Precision on the longitude =    0.05     deg

 --- Geomagnetic field model ---

      Model ( 0): DGRF 1995                          Epoch 1995.
                             Order + 1 =   11
                     Calculation epoch = 1995.0      year
         Colatitude of the dipole pole =   10.68     deg
          Longitude of the dipole pole =  -71.42     deg
                   Earth dipole moment =    0.302151 G/Re^3
          Correction for the SAA drift =    0.00     deg

 --- External magnetic field model ---

      Model ( 0): None                            
           Planetary activity index Kp =    0.0      (magnetopause check)

                                  Date =  01-Jan-1950
                        Universal Time =    0.00     hrs
            Greenwich angle of equinox =  100.08     deg
           GEO direction of the Sun, X =   -0.9200
                                     Y =   -0.0131
                                     Z =   -0.3918
                     Dipole tilt angle =  -25.91     deg

-------------------------
RESULT:
Third invariant =   -3.83261517738775E+07 ; Lstar =    2.07395076314524E+00
-------------------------
